system-addon/lib/Screenshots.jsm

```diff
@@ -73,9 +73,10 @@ this.Screenshots = {
    @ @param onScreenshot {function} Callback for when the screenshot loads
    */
   async maybeCacheScreenshot(link, url, property, onScreenshot) {
-    // Nothing to do if we already have a pending screenshot
+    // Nothing to do if we already have a pending screenshot or
+    // if a previous request failed and returned null.
     const cache = link.__sharedCache;
-    if (cache.fetchingScreenshot) {
+    if (cache.fetchingScreenshot || link[property] !== undefined) {
```

Comment:
I was thinking `=== null` but this is also a good sanity check if the link already has a screenshot, so it probably shouldn't fetch either.